modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_clock_zephyr.c

```diff
@@ -74,15 +74,19 @@ void nrf_802154_clock_hfclk_stop(void)
 
 void nrf_802154_clock_hfclk_start(void)
 {
+	int ret;
+
 	sys_notify_init_callback(&hfclk_cli.notify, hfclk_on_callback);
-	int ret = nrf_clock_control_request(DEVICE_DT_GET(DT_NODELABEL(hfxo)), NULL, &hfclk_cli);
+	ret = nrf_clock_control_request(DEVICE_DT_GET(DT_NODELABEL(hfxo)), NULL, &hfclk_cli);
 
 	__ASSERT_NO_MSG(ret >= 0);
```

Comment:
Don't you need instead to cast the variable to show that you intentionally don't use it when assertion checks are disabled?
```suggestion
	__ASSERT_NO_MSG(ret >= 0);
	(void)ret;
```

Reply:
I used a more 'lazy' approach to keep consistency in the file.
If I add some `UNUSED_ARG` macro, I will also have to adjust the other functions.

See e.g. line 62:
```c
int ret;
struct onoff_manager *mgr = 
        z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_HF);
```

Reply:
Sure, but I don't see how your change could solve the issue. When asserts are disabled the `ret` variable is still unused. Defining it in a separate statement doesn't make it used.